app/policies/activity_policy.rb

```diff
@@ -5,35 +5,54 @@ def index?
 
   def show?
     user.administrator? ||
-      fund? && user.fund_manager?
+      fund_and_fund_manager? ||
+      programme_and_fund_manager?
   end
 
   def create?
     user.administrator? ||
-      fund? && user.fund_manager?
+      fund_and_fund_manager? ||
+      programme_and_fund_manager?
   end
 
   def update?
     user.administrator? ||
-      fund? && user.fund_manager?
+      fund_and_fund_manager? ||
+      programme_and_fund_manager?
   end
 
   def destroy?
     user.administrator? ||
-      fund? && user.fund_manager?
+      fund_and_fund_manager? ||
+      programme_and_fund_manager?
   end
 
   private def fund?
     record.hierarchy.is_a?(Fund)
   end
 
+  private def programme?
+    record.hierarchy.is_a?(Programme)
```

Comment:
As we're not applying specific permissions only on either funds and programmes, how about replacing these with:

```ruby
private def hierarchy?
  record.hierarchy.is_a?(Programme)  || record.hierarchy.is_a?(Fund)
end
```

and then doing:

`user.adminstrator? || (user.fund_manager? && hierarchy?)`